holidays/countries/sierra_leone.py

```diff
@@ -0,0 +1,162 @@
+        2024: (6, 16),
+    }
+
+
+class SierraLeone(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
+    """Sierra Leone holidays.
+
+    References:
+        * <https://www.officeholidays.com/countries/sierra-leone>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Sierra_Leone>
+        * <https://www.timeanddate.com/holidays/sierra-leone/>
+    """
+
+    country = "SL"
+    default_language = "en_SL"
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("%s (observed, estimated)")
+    supported_languages = ("en_SL", "en_US")
+    # Sierra Leone gained independence on April 27, 1961.
+    start_year = 1962
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self, cls=SierraLeoneIslamicHolidays, show_estimated=False)
+        self._islamic_show_estimated = islamic_show_estimated
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_WORKDAY)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        dts_observed = set()
+
+        # New Year's Day.
+        dts_observed.add(self._add_new_years_day(tr("New Year's Day")))
+
+        if self._year >= 2002:
+            # Armed Forces Day.
+            dts_observed.add(self._add_holiday_feb_18(tr("Armed Forces Day")))
+
+        if self._year >= 2018:
+            # International Women's Day.
+            dts_observed.add(self._add_womens_day(tr("International Women's Day")))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        # Independence Day.
+        dts_observed.add(self._add_holiday_apr_27(tr("Independence Day")))
+
+        # International Worker's Day.
+        dts_observed.add(self._add_labor_day(tr("International Worker's Day")))
+
+        # Christmas Day.
+        dts_observed.add(self._add_christmas_day(tr("Christmas Day")))
+
+        # Boxing Day.
+        dts_observed.add(self._add_christmas_day_two(tr("Boxing Day")))
+
+        # Prophet's Birthday.
+        for dt in self._add_islamic_holiday(self._add_mawlid_day, tr("Prophet's Birthday")):
+            dts_observed.add(dt)
+
+        # Eid al-Fitr.
+        for dt in self._add_islamic_holiday(self._add_eid_al_fitr_day, tr("Eid al-Fitr")):
+            dts_observed.add(dt)
+
+        # Eid al-Adha.
+        for dt in self._add_islamic_holiday(self._add_eid_al_adha_day, tr("Eid al-Adha")):
+            dts_observed.add(dt)
```

Comment:
Why is this needed? Islamic holidays support should not differ from other countries.